src/main/java/org/wso2/lsp4intellij/IntellijLanguageClient.java

```diff
@@ -57,11 +58,12 @@
 
 public class IntellijLanguageClient implements ApplicationComponent, Disposable {
 
-    private static Logger LOG = Logger.getInstance(IntellijLanguageClient.class);
+    private static final Logger LOG = Logger.getInstance(IntellijLanguageClient.class);
     private static final Map<Pair<String, String>, LanguageServerWrapper> extToLanguageWrapper = new ConcurrentHashMap<>();
-    private static Map<String, Set<LanguageServerWrapper>> projectToLanguageWrappers = new ConcurrentHashMap<>();
-    private static Map<Pair<String, String>, LanguageServerDefinition> extToServerDefinition = new ConcurrentHashMap<>();
-    private static Map<String, LSPExtensionManager> extToExtManager = new ConcurrentHashMap<>();
+    private static final Map<String, Set<LanguageServerWrapper>> projectToLanguageWrappers = new ConcurrentHashMap<>();
+    private static final Map<Pair<String, String>, LanguageServerDefinition> extToServerDefinition = new ConcurrentHashMap<>();
+    private static final Map<Pair<Class<? extends FileType>, String>, LanguageServerDefinition> fileTypeToServerDefinition = new ConcurrentHashMap<>();
```

Comment:
I'm a bit new to this but it might be a safe assumption to assume that all file types are singletons and have this be
```
private static final Map<Pair<FileType, String>, LanguageServerDefinition> fileTypeToServerDefinition = new ConcurrentHashMap<>();
```